byterun/floats.c

```diff
@@ -553,34 +553,41 @@ CAMLprim value caml_copysign_float(value f, value g)
   return caml_copy_double(caml_copysign(Double_val(f), Double_val(g)));
 }
 
-CAMLprim value caml_eq_float(value f, value g)
-{
-  return Val_bool(Double_val(f) == Double_val(g));
-}
+#ifdef LACKS_SANE_NAN
 
-CAMLprim value caml_neq_float(value f, value g)
+CAMLprim caml_neq_float(value vf, value vg)
```

Comment:
Ugh - a refactoring typo, I expect. It's amazing (by which I mean horrifying!) that it still works!